### PR TITLE
ref 644499 - server command

### PR DESCRIPTION
**Pull request by aeter**
adds `stackit beta server command ...`

Testing Done:
tested locally. ran linters/formatters/tests/doc generators

### CODE REVIEW

File: internal/cmd/beta/server/command/create/create.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Creates a Server Command",
+		Long:  "Creates a Server Command.",
+		Args:  args.NoArgs,
+		Example: examples.Build(
+			examples.NewExample(
+				`Create a Server Command for server "xxx", with template name "RunShellScript" and a script from a file (using the @{...} format)`,
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/create/create.go:44]:
```suggestion
				`Create a server command for server with ID "xxx", template name "RunShellScript" and a script from a file (using the @{...} format)`,
```

File: internal/cmd/beta/server/command/create/create.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+				`$ stackit beta server command create --server-id xxx --template-name=RunShellScript --params script='@{/path/to/script.sh}'`),
+			examples.NewExample(
+				`Create a Server Command for server "xxx", with template name "RunShellScript" and a script provided on the command line`,
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/create/create.go:47]:
```suggestion
				`Create a server command for server with ID "xxx", template name "RunShellScript" and a script provided on the command line`,
```

File: internal/cmd/beta/server/command/create/create_test.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+		mod(&payload)
+	}
+	return payload
+}
+
+func TestParseInput(t *testing.T) {
+	tests := []struct {
+		description   string
+		flagValues    map[string]string
+		aclValues     []string
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/create/create_test.go:80]: Is aclValues really needed here?

File: internal/cmd/beta/server/command/describe/describe.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("describe %s", commandIdArg),
+		Short: "Shows details of a Server Command",
+		Long:  "Shows details of a Server Command.",
+		Args:  args.SingleArg(commandIdArg, nil),
+		Example: examples.Build(
+			examples.NewExample(
+				`Get details of a Server Command with id "zzz" for server "xxx"`,
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/describe/describe.go:41]:
```suggestion
				`Get details of a Server Command with ID "xxx" for server with ID "yyy"`,
```

File: internal/cmd/beta/server/command/describe/describe.go [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+				"$ stackit beta server command describe zzz --server-id=xxx"),
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/describe/describe.go:42]:
```suggestion
				"$ stackit beta server command describe xxx --server-id=yyy"),
```

File: internal/cmd/beta/server/command/describe/describe.go [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+			examples.NewExample(
+				`Get details of a Server Command with id "zzz" for server "xxx" in JSON format`,
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/describe/describe.go:44]:
```suggestion
				`Get details of a Server Command with ID "xxx" for server with ID "yyy" in JSON format`,
```

File: internal/cmd/beta/server/command/describe/describe.go [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+				"$ stackit beta server command describe zzz --server-id=xxx -o json"),
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/describe/describe.go:45]:
```suggestion
				"$ stackit beta server command describe xxx --server-id=yyy --output-format json"),
```

File: internal/cmd/beta/server/command/describe/describe_test.go [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+		},
+		{
+			description: "project id invalid 2",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = "invalid-uuid"
+			}),
+			isValid: false,
+		},
+	}
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/describe/describe_test.go:127]: I think here we can add 4 more tests cases

File: internal/cmd/beta/server/command/list/list.go [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+		details, err := yaml.MarshalWithOptions(commands, yaml.IndentSequence(true))
+		if err != nil {
+			return fmt.Errorf("marshal server command list: %w", err)
+		}
+		p.Outputln(string(details))
+
+		return nil
+	default:
+		table := tables.NewTable()
+		table.SetHeader("ID", "STATUS", "STARTED_AT", "FINISHED_AT", "TEMPLATE NAME", "TEMPLATE TITLE")
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/list/list.go:149]: Maybe you can change the order of the columns, generally we add name and title after id directly. 

Same comment also for the describe command output :)

File: internal/cmd/beta/server/command/template/describe/describe.go [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("describe %s", commandTemplateNameArg),
+		Short: "Shows details of a Server Command Template",
+		Long:  "Shows details of a Server Command Template.",
+		Args:  args.SingleArg(commandTemplateNameArg, nil),
+		Example: examples.Build(
+			examples.NewExample(
+				`Get details of a Server Command Template with name "RunShellScript" for server "xxx"`,
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/template/describe/describe.go:42]:
```suggestion
				`Get details of a Server Command Template with name "RunShellScript" for server with ID "xxx"`,
```

File: internal/cmd/beta/server/command/template/describe/describe.go [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+				"$ stackit beta server command template describe RunShellScript --server-id=xxx"),
+			examples.NewExample(
+				`Get details of a Server Command Template with name "RunShellScript" for server "xxx" in json format`,
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/template/describe/describe.go:45]:
```suggestion
				`Get details of a Server Command Template with name "RunShellScript" for server with ID "xxx" in JSON format`,
```

File: internal/cmd/beta/server/command/template/describe/describe.go [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+				"$ stackit beta server command template describe RunShellScript --server-id=xxx -o json"),
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/template/describe/describe.go:46]:
```suggestion
				"$ stackit beta server command template describe RunShellScript --server-id=xxx --output-format json"),
```

File: internal/cmd/beta/server/command/template/describe/describe_test.go [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+			isValid: false,
+		},
+		{
+			description: "project id invalid 2",
+			argValues:   fixtureArgValues(),
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = "invalid-uuid"
+			}),
+			isValid: false,
+		},
```
Comment (GokceGK) [inline internal/cmd/beta/server/command/template/describe/describe_test.go:126]: Here we can also add extra test cases 

		{
			description: "server id missing",
			argValues:   fixtureArgValues(),
			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
				delete(flagValues, serverIdFlag)
			}),
			isValid: false,
		},
		{
			description: "server id invalid 1",
			argValues:   fixtureArgValues(),
			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
				flagValues[serverIdFlag] = ""
			}),
			isValid: false,
		},
		{
			description: "server id invalid 2",
			argValues:   fixtureArgValues(),
			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
				flagValues[serverIdFlag] = "invalid-uuid"
			}),
			isValid: false,
		},
		{
			description: "command template name invalid 1",
			argValues:   []string{""},
			flagValues:  fixtureFlagValues(),
			isValid:     false,
		},

File: internal/cmd/config/set/set.go [hunk 14]
```diff
@@ -144,6 +145,7 @@ func configureFlags(cmd *cobra.Command) {
 	cmd.Flags().String(serviceAccountCustomEndpointFlag, "", "Service Account API base URL, used in calls to this API")
 	cmd.Flags().String(serviceEnablementCustomEndpointFlag, "", "Service Enablement API base URL, used in calls to this API")
 	cmd.Flags().String(serverBackupCustomEndpointFlag, "", "Server Backup API base URL, used in calls to this API")
+	cmd.Flags().String(runCommandCustomEndpointFlag, "", "Runcommand API base URL, used in calls to this API")
```
Comment (GokceGK) [inline internal/cmd/config/set/set.go:148]:
```suggestion
	cmd.Flags().String(runCommandCustomEndpointFlag, "", "Run Command API base URL, used in calls to this API")
```

File: internal/cmd/beta/server/command/describe/describe_test.go (hunk 8 above)
Comment (aeter) [reply internal/cmd/beta/server/command/describe/describe_test.go:127]: Very nice! Thank you!

File: internal/cmd/beta/server/command/template/describe/describe_test.go (hunk 13 above)
Comment (aeter) [reply internal/cmd/beta/server/command/template/describe/describe_test.go:126]: Awesome! Thank you!

**Review by GokceGK — APPROVED** [review]

**Pull request merged**